lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -131,11 +140,25 @@ import qualified Network.Wai.Handler.Warp as Warp
 
 -- | Start the application server
 start
+    :: forall t. (TxId t, KeyToAddress t, EncodeAddress t, DecodeAddress t)
+    => (Warp.Port -> IO ())
+    -> Maybe Warp.Port
+    -> WalletLayer (SeqState t) t
+    -> IO ()
+start logStartup mport wl =
```

Comment:
Minor remark: `logStartup` could become misleading as, the name implies the action. But the action itself is left to the caller. `onStartup` would be less confusing -- we never know what the caller will yield as `IO` in the end :sweat_smile: